test/plugin/test_out_google_cloud.rb

```diff
@@ -240,10 +244,16 @@ def http_request_message
     HTTP_REQUEST_MESSAGE
   end
 
-  # A wrapper around the constant HTTP_REQUEST_MESSAGE_WITHOUT_REFERER, so the
-  # definition can be skipped in the shared module and defined here.
-  def http_request_message_without_referer
-    HTTP_REQUEST_MESSAGE_WITHOUT_REFERER
+  # A wrapper around the constant HTTP_REQUEST_MESSAGE_EXPECTED_WITHOUT_REFERER,
+  # so the definition can be skipped in the shared module and defined here.
+  def http_request_message_expected_without_referer
```

Comment:
I think we've reached an unwarranted level of complexity here. Let's do this: keep only `http_request_message` as a virtual function (with the corresponding constant), add a non-virtual `http_request_message_with_nil_referer` that just returns `http_request_message.merge('referer' => nil)`, and move the special handling of the absent referer into the grpc test. In fact, you could even move the `HTTP_REQUEST_MESSAGE` constant into the base class, and override `http_request_message` in this file to rename the key.